src/MinHash.java

```diff
@@ -1,98 +1,59 @@
 import java.io.*;
+import java.nio.file.Files;
 import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class MinHash {
-
    int numPerm;
-   String folderName;
-   int[][] mhMatrix;
-   int[][] tdMatrix;
-   int numTerms;
-   String[] termMatrix;
-   String[] docs;
-
-
-
-    public MinHash(String folder, int numPermutations){
+   MinHashMatrix minHashMatrix;
+   TermDocumentMatrix termDocumentMatrix;
 
-        File dir = new File("src/"+folder);
-        File [] files = dir.listFiles();
-        String docs[] = new String[files.length];
-        ArrayList<String> Terms = new ArrayList<String>();
-        ArrayList<String>[] Dterms = new ArrayList[files.length];
+    public MinHash(String folder, int numPermutations) {
+        termDocumentMatrix = new TermDocumentMatrix();
+        numPerm = numPermutations;
+        File dir = new File("src/" + folder);
+        File[] files = dir.listFiles();
 
-        for(int i=0;i< docs.length;i++){
-            docs[i]=files[i].getName();
-        }
-        for (int i =0; i< files.length;i++){
-            Dterms[i] = new ArrayList<String>();
-            try {
-                File myObj = new File(files[i].getAbsolutePath());
-                Scanner myReader = new Scanner(myObj);
-                while (myReader.hasNextLine()) {
-                    String data = myReader.nextLine();
-                    String[] terms = data.split("[,|.$':; ]");
-                    for (int j=0; j< terms.length;j++){
-                        Terms.add(terms[j]);// adding term to set of all terms
-                        Dterms[i].add(terms[j]);// adding term to set of terms in particular document i
-                    }
+        createTermDocumentMatrix(files);
+        minHashMatrix = new MinHashMatrix(termDocumentMatrix, numPermutations);
+    }
 
-                    //System.out.println(data);
+    private void createTermDocumentMatrix(File[] files) {
+        if (files != null) {
+            for (File file : files) {
+                try (Stream<String> stream = Files.lines(file.toPath())) {
+                    stream.map(line -> {
```

Comment:
This code does the following: for each line in the file, split it by the punctuation. Then we filter out terms whose length is greater than 2 and terms who are not equal to "the" and collect it as a pre processed list. Then we pass this list and the terms to the TermDocumentMatrix class to add it to the term document matrix.